src/fuzzfetch/fetch.py

```diff
@@ -76,7 +76,11 @@ def __init__(self, revision, branch):
         """
         if branch is None or branch == '?':
             raise FetcherException("Can't lookup revision date for branch: %r" % (branch,))
-        if branch != 'try':
+        if branch == 'try':
+            pass
```

Comment:
I think it would be clearer to see the last `else:` keep the inverted condition.